### PR TITLE
Drop support for python 2.7 and old versions of sphinx

### PR DESCRIPTION
**Pull request by stsewd**
And start testing new versions

### CODE REVIEW

**Review by Blendify — COMMENTED** [review]
What about dropping all of 1.x?

File: tox.ini [hunk 1]
```diff
@@ -1,10 +1,10 @@
 [tox]
-envlist = py{27,36,37}-sphinx{16,17,18,20,21,22,23,24,30}
+envlist = py{36,37,38}-sphinx{18,20,21,22,23,24,30,31,32}
```
Comment (humitos) [inline tox.ini:2]: We could add 3.9 as well here that was just released.

**Review by humitos — COMMENTED** [review]
I'm fine deprecating Python 2.x versions and Sphinx 1.x. However, I wouldn't do it just yet if it's not causing us any troubles. I'm sure there are many projects still using Python 2.x and/or Sphinx 1.8 (BTW, our current default in RTD) and they won't be able to upgrade to the new theme versions with new features.

So, I would only deprecate them if it's blocking us to build more features or it it's costing us extra effort to write the same code in different ways to support Python 2.x

**Comment by jdillard** [comment]
Would it be worth mentioning plans to deprecate sphinx/python support in version X of the theme? So people have time/incentive to make the switch?

**Comment by stsewd** [comment]
> Would it be worth mentioning plans to deprecate sphinx/python support in version X of the theme? So people have time/incentive to make the switch?

So, we decided to not deprecate this just now, but we are planning to have a page in rtd with the supported versions and EOL

File: tox.ini (hunk 1 above)
Comment (stsewd) [reply tox.ini:2]: Isn't in travis yet https://docs.travis-ci.com/user/languages/python/#python-versions

**Comment by stsewd** [comment]
Closing in favor of https://github.com/readthedocs/sphinx_rtd_theme/pull/1012

**Pull request closed without merging**